### PR TITLE
Add updated widget targets

### PR DESCRIPTION
**Pull request by GerardasB**
This PR adds redesigned widget targets (with increased discoverability of target areas, since the targets are displayed once the drag interaction is started).
For now this is an internal opt-in feature, which can be enabled by wrapping an application in a `TargetOptionsContext`.
`ui-test-app` has a toggle in a `Settings` page to quickly change between an old and a new target designs.

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

File: test-apps/ui-test-app/src/frontend/appui/frontstages/Settings.tsx [hunk 1]
```diff
@@ -52,3 +54,18 @@ function SettingsItem(props: SettingsItemProps) {
     </div>
   );
 }
+
+export function TargetSettingsPageComponent() {
```
Comment (NancyMcCallB) [inline test-apps/ui-test-app/src/frontend/appui/frontstages/Settings.tsx:58]: Could you add this to the providers for the new appui-test-providers package?

Comment (GerardasB) [reply test-apps/ui-test-app/src/frontend/appui/frontstages/Settings.tsx:58]: Enabled v2 targets in appui-test-app.

**Review by raplemie — COMMENTED** [review]
Nice work! I was if we shouldnt emphasis in `DropTarget` instead of simply `Target` in the names, as target seems quite generic out of context, but I've seen that it was already simply named Target before. (For example, there is TabDropTargetState which is a combination of many xxxTargetState, are these all Drop targets or they can be other type of targets (Click target, or simply modification target of something...)

File: ui/appui-layout-react/src/appui-layout-react/base/DragManager.tsx [hunk 2]
```diff
@@ -6,28 +6,30 @@
  * @module Base
  */
 import * as React from "react";
+import * as _ from "lodash";
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-layout-react/src/appui-layout-react/base/DragManager.tsx:9]: It looks like **lodash/core** is missing from `@types/lodash` definitions and to import the `lodash/isEqual` directly I'd have to enable `esModuleInterop`, which messes up some of our flaky tests. Let's address this in a separate PR.

File: ui/appui-layout-react/src/appui-layout-react.ts [hunk 3]
```diff
@@ -64,6 +64,8 @@ export * from "./appui-layout-react/stage-panels/manager/NestedStagePanels";
 export * from "./appui-layout-react/stage-panels/manager/StagePanel";
 export * from "./appui-layout-react/stage-panels/manager/StagePanels";
 
+export * from "./appui-layout-react/target/TargetOptions";
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-layout-react/src/appui-layout-react.ts:67]: You are right, we are exposing this as an internal opt-in feature, which we will enable by default in the future.

**Pull request merged**